src/Akeneo/Bundle/ElasticsearchBundle/Cursor/BoundedCursorFactory.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Akeneo\Bundle\ElasticsearchBundle\Cursor;
+
+use Akeneo\Bundle\ElasticsearchBundle\Client;
+use Akeneo\Component\StorageUtils\Cursor\CursorFactoryInterface;
+use Akeneo\Component\StorageUtils\Exception\InvalidObjectException;
+use Akeneo\Component\StorageUtils\Repository\CursorableRepositoryInterface;
+use Doctrine\Common\Persistence\ObjectManager;
+
+/**
+ * Cursor factory to instantiate an elasticsearch bounded cursor
+ *
+ * @author    Marie Bochu <marie.bochu@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class BoundedCursorFactory implements CursorFactoryInterface
+{
+    /** @var Client */
+    protected $searchEngine;
+
+    /** @var ObjectManager */
+    protected $om;
+
+    /** @var string */
+    protected $entityClassName;
+
+    /** @var string */
+    protected $cursorClassName;
+
+    /** @var int */
+    protected $pageSize;
+
+    /** @var string */
+    protected $indexType;
+
+    /**
+     * @param Client        $searchEngine
+     * @param ObjectManager $om
```

Comment:
I'd like to avoid to have a reference to ORM, that's why I prefer to have ObjectManager (which is in Commong namespace)